src/components/RouterBreadcrumbs.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import Breadcrumbs from '@material-ui/lab/Breadcrumbs';
+import Link from '@material-ui/core/Link';
+import Typography from '@material-ui/core/Typography';
+import { Link as RouterLink } from 'react-router-dom';
+import { Paper, withStyles } from '@material-ui/core';
+
+const styles = (theme) => ({
+  paper: {
+    padding: `${theme.spacing.unit}px ${theme.spacing.unit * 2}px`,
+  },
+  reactRouterHoverInherit: theme.reactRouterHoverInherit
+});
+
+/**
+ * Returns a list of breadcrumbs based on current path of navigation
+ * @param {string} pathname current URI pathname
+ * @return {Array} List of BreadcrumbItems
+ */
+const RouterBreadcrumbs = ({ pathname, classes }) => {
+  const breadcrumbItems = [];
+  const viewNames = ['browser', 'viewer', 'editor', 'repos'];
+  const keyBase = 'breadcrumb-item-';
+
+  const subDirectoriesArr = pathname.split('/');
+
+  // pathname starts with '/', so discard. Pathname will sometimes end with '/', so discard that as well.
+  subDirectoriesArr.shift();
+  if (subDirectoriesArr[subDirectoriesArr.length - 1] === '') {
+    subDirectoriesArr.pop();
+  }
+
+
+  let currDir = '';
+  for (let i = 0; i < subDirectoriesArr.length; i++) {
+    const directory = subDirectoriesArr[i];
+    currDir = currDir.concat('/', directory);
+
+    if (viewNames.indexOf(directory) !== -1) {
```

Comment:
This introduces a bug where breadcrumbs doesn't work correctly if the user has a folder in their repository with any of browser, viewer, editor, repos.

Reply:
👍 changing this to index based, since our URL shouldn't be changing.